src/main/java/com/iridium/iridiumskyblock/managers/SchematicManager.java

```diff
@@ -115,4 +125,78 @@ private CompletableFuture<Void> pasteSchematic(Island island, Schematics.Schemat
         return completableFuture;
     }
 
+    public boolean processTransaction(Player player, Schematics.SchematicConfig schematic) {
+
+        User user = IridiumSkyblock.getInstance().getUserManager().getUser(player);
+        Optional<Island> island = IridiumSkyblock.getInstance().getTeamManager().getTeamViaID(user.getTeamID());
+
+        if(!island.isPresent()){
+            //NO ISLAND
+            return false;
+        }
+
+        TeamBank bank;
+
+        double bankBalance;
+        List<Placeholder> bankPlaceholders;
+        double moneyCost = schematic.regenCost.money;
+        Economy economy = IridiumSkyblock.getInstance().getEconomy();
+
+        if(moneyCost == 0 || economy != null && economy.getBalance(player) >= moneyCost) {
+            //INVALID TRANSACTION
+            return false;
+        }
```

Comment:
I think this needs to be flipped?

In the origional code this was if the player can afford to buy the item. So we need to flip it and return false if they cant afford to buy it?

```suggestion
        if(moneyCost != 0 && economy != null && economy.getBalance(player) < moneyCost) {
            //INVALID TRANSACTION
            return false;
        }
```